### PR TITLE
Add changelog enforcement.

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Make sure we have a changelog entry
+  changelog:
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dangoslen/changelog-enforcer@v1.4.0
```
Comment (rpdome) [inline .github/workflows/pull-request.yml:20]: Have we got (do we need) an approval for this?

Comment (rpdome) [inline .github/workflows/pull-request.yml:20]: Have we got (do we need) an approval for these two?

**Comment by rpdome** [comment]
Thanks for doing this. once this is checked in, would you mind adding the same thing to common, Broker, ADAL as well?

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (rpdome) [reply .github/workflows/pull-request.yml:20]: The [OSS guide](https://docs.opensource.microsoft.com/content/using/index.html) seems to be focusing on open source components that are part of the product only. I'd suggest reaching out to them to see if there's anything we have to do in order to use it as part of our build pipeline.

**Comment by AdamBJohnsonx** [comment]
> Thanks for doing this. once this is checked in, would you mind adding the same thing to Broker (and ADAL?) as well?

I did that.  Ideas to reach out to on OSS, though?

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (AdamBJohnsonx) [reply .github/workflows/pull-request.yml:20]: There doesn't seem to be any guidance right now.  Eventually they'll end up needing to have a path for these, but right now there's nothing.  I'm thinking we use it now, and clone it if we need to.

**Review by iambmelt — APPROVED** [review]

**Pull request closed without merging**